### PR TITLE
[Travis] Continuous integration for each Sylius package included in the build

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #4408
| License         | MIT
| Doc PR          | -

- [x] Set `prefer-stable` to `true` in all packages' `composer.json`
- [x] Updated Sylius packages versioning from different formats (mostly `0.18.*`, but there was `0.18.*@dev` and `^0.18.0` too) to simple`^0.18`)
- [x] Removed vertical alignment in `composer.json`
- [x] Fixed not installable packages
- [x] Added dependencies downloading for each package
- [x] Added Composer validation for each package
- [x] Added isolated tests running for each package (phpspec & phpunit support)
- [x] Added caching for packages' `composer.lock` (only the first builds will take much more time :tada:)

### CODE REVIEW

**Comment by bendavies** [comment]
You could run all your shell scripts through http://www.shellcheck.net/

File: etc/concurrent.lib.sh [hunk 1]
```diff
@@ -0,0 +1,1067 @@
...
+        echo '>>> DRY RUN (concurrent): The "$CONCURRENT_DRY_RUN" environment variable is set. <<<'
+    fi
+
+    __crt__hide_cursor
+    __crt__start_all_tasks
+    __crt__wait_for_all_tasks
+    __crt__show_cursor
+
+    exit ${__crt__final_status}
+)
```
Comment (pjedrzejewski) [inline etc/concurrent.lib.sh]: Any chance we can avoid committing that? Downloading?

Comment (pamil) [inline etc/concurrent.lib.sh]: Yeah, I've just get rid of that in favour of GNU parallel :tada:

File: src/Sylius/Bundle/AdminBundle/composer.json [hunk 2]
```diff
@@ -7,23 +7,23 @@
...
             "homepage": "http://github.com/Sylius/Sylius/contributors"
         }
     ],
     "require": {
         "php": "^5.5.9|^7.0",
 
-        "sylius/ui-bundle": "^0.17.0",
-        "sylius/core-bundle": "^0.17.0",
+        "sylius/ui-bundle": "^0.18",
+        "sylius/core-bundle": "^0.18",
```
Comment (liverbool) [inline src/Sylius/Bundle/AdminBundle/composer.json]: should be `0.18.*`?

**Comment by pjedrzejewski** [comment]
Thanks Kamil! Awesome work! :+1: This will help a lot...

**Pull request merged**